toml.py

```diff
@@ -793,15 +794,16 @@ def dumps(o, preserve=False):
                     retval += addtoretval
             for s in addtosections:
                 newsections[section + "." + s] = addtosections[s]
+            retval += '\n'
         sections = newsections
-    return retval
+    return retval[:-1]
 
 
-def _dump_sections(o, sup, preserve=False):
+def _dump_sections(o, sup, preserve=False, _dict=dict):
     retstr = ""
     if sup != "" and sup[-1] != ".":
         sup += '.'
-    retdict = {}
+    retdict = _dict()
```

Comment:
Wouldn't it make more sense to just use `o.__class__()` here?

Reply:
Yes. That will solve the use case. I'll open a PR with that change.

Reply:
Hi @amjith , is this change moving along? Do you need any help with it?

Reply:
I've already made a PR #121 that solves this issue. It has been merged. 

This PR can be closed.